dpnp/dparray.pyx

```diff
@@ -788,19 +790,21 @@ cdef class dparray:
 
         """
 
-        if casting is not None:
-            utils.checker_throw_value_error("astype", "casting", casting, None)
-
-        if subok is not None:
-            utils.checker_throw_value_error("astype", "subok", subok, None)
-
-        if copy is not True:
-            utils.checker_throw_value_error("astype", "copy", copy, True)
-
-        if order is not 'K':
-            utils.checker_throw_value_error("astype", "order", order, 'K')
+        if (not use_origin_backend(self)):
+            if not isinstance(self, dparray):
+                pass
+            elif casting is not None:
+                pass
+            elif subok is not None:
+                pass
+            elif copy is not True:
+                pass
+            elif order is not 'K':
+                pass
+            else:
+                return dpnp_astype(self, dtype)
 
-        return dpnp_astype(self, dtype)
+        return call_origin(numpy.ndarray.astype, self, dtype, order, casting, subok, copy)
```

Comment:
I think doing fallback on `numpy.ndarray` here is bad idea.
It is better just throw error.